pkg/virtual/framework/forwardingregistry/rest.go

```diff
@@ -44,32 +45,38 @@ func newStores(ctx context.Context, gvr schema.GroupVersionResource, dynamicClus
 			patchConflictRetryBackoff = &retry.DefaultRetry
 		}
 
-		store := &Store{
-			NewFunc: func() runtime.Object {
-				// set the expected group/version/kind in the new object as a signal to the versioning decoder
-				ret := &unstructured.Unstructured{}
-				ret.SetGroupVersionKind(kind)
-				return ret
-			},
-			NewListFunc: func() runtime.Object {
-				// lists are never stored, only manufactured, so stomp in the right kind
-				ret := &unstructured.UnstructuredList{}
-				ret.SetGroupVersionKind(listKind)
-				return ret
-			},
-			DefaultQualifiedResource: resource,
-			CreateStrategy:           strategy,
-			UpdateStrategy:           strategy,
-			DeleteStrategy:           strategy,
-			TableConvertor:           tableConvertor,
-			ResetFieldsStrategy:      strategy,
+		requirements, selectable := labels.SelectorFromSet(labels.Set(labelSelector)).Requirements()
+		if !selectable {
+			// why are you even asking for this store without a selector that can match anything? error
```

Comment:
There's no way to bubble an error out of the `apiserver.RestProviderFunc` ... want me to refactor that?

Reply:
if everything is static, a panic is ok'ish.

Reply:
would go with `labels.Selector` as the type you pass in here. Then you can do all the error-prone calls here: https://github.com/kcp-dev/kcp/blob/main/pkg/virtual/syncer/builder/build.go#L116